src/gcodeExport.cpp

```diff
@@ -57,7 +57,23 @@ void GCodeExport::preSetup(const MeshGroup* meshgroup)
 
     for (unsigned int extruder_nr = 0; extruder_nr < extruder_count; extruder_nr++)
     {
-        const ExtruderTrain* train = meshgroup->getExtruderTrain(extruder_nr);
+        ExtruderTrain* train = meshgroup->getExtruderTrain(extruder_nr);
+
+        // Cura only has the global stack when it is a single-extrusion machine.
+        // In this case, wee need to update the extruder train settings according to the global stack.
+        if (extruder_count == 1 and extruder_nr == 0)
+        {
+            for (auto setting_key : train->getAllLocalSettingKeys())
+            {
+                if (not storage.hasSettingKey(setting_key))
+                {
+                    continue;
+                }
+
+                train->setSetting(setting_key, storage.getSettingString(setting_key));
+            }
+        }
```

Comment:
`MeshGroup.createExtruderTrain()` is a nicer place for this piece of code, but it's not an easy task to move it there. A major refactoring might be required.